selinux/cephmetrics.te

```diff
@@ -31,15 +31,13 @@ allow collectd_t bin_t:file { execute execute_no_trans };
 allow collectd_t ceph_t:unix_stream_socket connectto;
 allow collectd_t ceph_exec_t:file { execute execute_no_trans open read };
 allow collectd_t ceph_var_run_t:dir read;
-allow collectd_t ceph_var_lib_t:file { getattr open read };
```

Comment:
Why did you remove these? It is very likely we will still need them. This gives collectd a basic read+stat access for ceph files which we will probably still need.

Just because you get a clean run where you don't need some of the rules, it does not mean they are not necessary in certain cases.

Reply:
I had removed the specific allow rule, since the policy libs_manage_lib policy is enabled for collectd - which obviously encompasses the osd directories under /var/lib/ceph.

However, perhaps we should try and avoid this ping-pong. I'm happy for you to propose an updated policy that is tested and works - please just push a commit to this PR.

I'd like to get QE testing enabled, and without an updated policy QE is blocked.

Reply:
It is not as obvious as it may seem, `/var/lib` and `/var/lib/ceph` have different contexts and as such, the method could potentially encompass `/var/lib/ceph` but it is rather unlikely to do so as the context for `/var/lib/ceph` is defined in ceph policy which the base targeted policy that defines `libs_manage_lib_*` does not know about.

The safest thing you can do is to simply add new rules without removing the old ones. That way, you can have a working policy asap.

btw: QE does not need to be blocked by this. All they need to do is to run their tests with SELinux in Permissive mode. They will still get the avc denials that they can report back and the tests would continue to work as if there were no denials.